Misc/NEWS.d/next/Library/2025-04-08-14-50-39.gh-issue-127495.Q0V0bS.rst

```diff
@@ -0,0 +1,3 @@
+Save (append new entry) the repl history to file every time new statement
+was succesful.  This should preserve command-line history after interpreter
+crash. Patch by Sergey B Kirpichev.
```

Comment:
"Successful" is misleading here (and a typo). In your test `time.sleep()` doesn't successfully finish executing, but it's put in history already.

```suggestion
In PyREPL, append a new entry to the ``PYTHON_HISTORY`` file after every
statement. This should preserve command-line history after interpreter
is terminated. Patch by Sergey B Kirpichev.
```

Reply:
> a typo

That's true :(

> In your test time.sleep() doesn't successfully finish executing, but it's put in history already.

No!  It shouldn't and added test checks that.